cprover_bindings/src/goto_program/expr.rs

```diff
@@ -282,13 +282,44 @@ impl Expr {
             _ => false,
         }
     }
+
+    /// Returns whether an expression causes side effects or not
     pub fn is_side_effect(&self) -> bool {
-        match *self.value {
+        /// Returns whether any of the expressions on a vector
+        /// causes side effects or not
+        fn are_side_effect(exprs: &Vec<Expr>) -> bool {
+            for expr in exprs {
+                if expr.is_side_effect() {
+                    return true;
+                }
+            }
+            false
+        }
```

Comment:
Style ⛏️ : you could use `exprs.iter().any(|e| e.is_side_effect())`

Reply:
Thanks!